swap_meet/vendor.py

```diff
@@ -25,35 +24,41 @@ def get_by_id(self, item_id):
     
     def swap_items(self, other_vendor, my_item, their_item):
 
-        if my_item in self.inventory and their_item in other_vendor.inventory:
-            self.inventory.remove(my_item) 
-            self.inventory.append(their_item)
-            other_vendor.inventory.remove(their_item)
-            other_vendor.inventory.append(my_item)
-            return True
-        
-        return False
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+            
+        self.inventory.remove(my_item) 
+        self.inventory.append(their_item)
+        other_vendor.inventory.remove(their_item)
+        other_vendor.inventory.append(my_item)
+        return True
+
     
     def swap_first_item(self, other_vendor):
+
         if not self.inventory or not other_vendor.inventory:
             return False
         
-        elif self.inventory and other_vendor.inventory:
-            self.inventory.append(other_vendor.inventory[0])
-            other_vendor.inventory.append(self.inventory[0])
-            self.inventory.remove(self.inventory[0]) 
-            other_vendor.inventory.remove(other_vendor.inventory[0])
-            return True
+        self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
```

Comment:
👍 Once we've confirmed that each vendor has items to swap, we can use our previous `swap_items`.

Alternatively, we could consider `swap_first_item` as being a function where we have additional information about the swap being attempted. We could provide a custom implementation if there was a difference in approach that would benefit the performance of this method (i.e., how `swap_items` is O(n), but we might haev an approach that is O(1) for this scenario!).

Yet another optiuon would be to try to implement `swap_items` in such a way that it remains no worse than On) in the general case, but if used as a helper for this method, could be predictably analyzed to be O(1).